[Docs] Improve initial pages (#401)
pasqal-io/qadence: docs/content/parameters.md

- Feature parameter: a non-trainable parameter which can be used to pass input values.

## Fixed parameters

Passing fixed parameters to blocks can be done by simply passing a Python numeric type or a `torch.Tensor`.

```python exec="on" source="material-block" result="json"
import torch
from qadence import RX, run, PI

wf = run(RX(0, torch.tensor(PI)))
print(f"{wf = }") # markdown-exec: hide

wf = run(RX(0, PI))
print(f"{wf = }") # markdown-exec: hide
```

## Variational parameters

To parametrize a block a `VariationalParameter` instance is required. In most cases Qadence also accepts a Python string, which will be used to automatically initialize a `VariationalParameter`:

```python exec="on" source="material-block" result="json"
from qadence import RX, run, VariationalParameter

block = RX(0, VariationalParameter("theta"))
block = RX(0, "theta")  # Equivalent

wf = run(block)
print(f"{wf = }") # markdown-exec: hide
```

By calling `run`, a random value for `"theta"` is initialized at execution. In a `QuantumModel`, variational parameters are stored in the underlying model parameter dictionary.

## Feature parameters

> A `FeatureParameter` type can also be used, which will require an input value or a batch of values. In most cases, Qadence accepts a `values` dictionary to set the input of feature parameters.

Reviewer: ```suggestion
A `FeatureParameter` type can also be used. It requires an input value or a batch of values. In most cases, Qadence accepts a `values` dictionary to set the input of feature parameters.
```
